gateway-ha/src/main/java/io/trino/gateway/ha/domain/RoutingRules.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.gateway.ha.domain;
+
+import com.google.common.collect.ImmutableList;
+import jakarta.annotation.Nullable;
+
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * RoutingRules
+ *
+ * @param name name of the routing rule
+ * @param description description of the routing rule
+ * @param priority priority of the routing rule. Higher number represents higher priority. If two rules have same priority then order of execution is not guaranteed.
+ * @param actions actions of the routing rule
+ * @param condition condition of the routing rule
+ */
+public record RoutingRules(
```

Comment:
this is a single rule
```suggestion
public record RoutingRule(
```